models/MARBL_joint_estimation/model_mod.f90

```diff
@@ -0,0 +1,654 @@
+                             get_location, VERTISLEVEL
+
+use    utilities_mod, only : register_module, error_handler, &
+                             E_ERR, E_MSG, &
+                             nmlfileunit, do_output, do_nml_file, do_nml_term,  &
+                             find_namelist_in_file, check_namelist_read, &
+                             to_upper
+
+use netcdf_utilities_mod, only : nc_add_global_attribute, nc_synchronize_file, &
+                                 nc_add_global_creation_time, &
+                                 nc_begin_define_mode, nc_end_define_mode, &
+                                 NF90_MAX_NAME, nc_open_file_readonly, &
+                                 nc_get_variable, nc_get_variable_size, nc_close_file
+
+use state_structure_mod, only : add_domain, get_domain_size, get_model_variable_indices, &
+                                get_varid_from_kind, get_dart_vector_index
+
+use obs_kind_mod, only : get_index_for_quantity, QTY_U_CURRENT_COMPONENT, &
+                         QTY_V_CURRENT_COMPONENT, QTY_DRY_LAND, &
+                         QTY_LAYER_THICKNESS
+
+use distributed_state_mod, only: get_state
+
+use ensemble_manager_mod, only : ensemble_type
+
+! These routines are passed through from default_model_mod.
+! To write model specific versions of these routines
+! remove the routine from this use statement and add your code to
+! this the file.
+use default_model_mod, only : pert_model_copies, write_model_time, &
+                              init_time => fail_init_time, &
+                              init_conditions => fail_init_conditions, &
+                              convert_vertical_obs, convert_vertical_state, adv_1step
+
+implicit none
+private
+
+! routines required by DART code - will be called from filter and other
+! DART executables. 
+public :: get_model_size,         &
+          get_state_meta_data,    &
+          model_interpolate,      &
+          end_model,              &
+          static_init_model,      &
+          nc_write_model_atts,    &
+          get_close_obs,          &
+          get_close_state,        &
+          pert_model_copies,      &
+          convert_vertical_obs,   &
+          convert_vertical_state, &
+          read_model_time,        &
+          adv_1step,              &
+          init_time,              &
+          init_conditions,        &
+          shortest_time_between_assimilations, &
+          write_model_time
+
+
+character(len=256), parameter :: source = "model_mod.f90"
+logical :: module_initialized = .false.
+integer :: state_dom_id  ! used to access the state structure
+integer :: param_dom_id  ! used to access MARBL internal parameters
+integer :: nfields ! number of fields in the state or parameter vector
+integer :: nz      ! the number of vertical layers
+integer :: model_size
+type(time_type) :: assimilation_time_step
+real(r8), parameter :: geolon = 360 - 64.0
+real(r8), parameter :: geolat = 31.0
```

Comment:
these are hardcoded, should these be namelist options for people to put the column anywhere on Earth?

Reply:
I don't see any reason to not make them namelist options, but a user will need to know that any mismatch between the location in the obs-seq file and `geolat`/`geolon` will cause DART to reject the observation. I think that was our motive for hard-coding those coordinates. We wanted to make sure that they always agreed with the coordinates written in the obs-seq file, which in turn are hard-coded into the obs converter.